modules/bullet/bullet_physics_server.cpp

```diff
@@ -408,13 +408,19 @@ Transform BulletPhysicsServer::area_get_transform(RID p_area) const {
 void BulletPhysicsServer::area_set_collision_mask(RID p_area, uint32_t p_mask) {
 	AreaBullet *area = area_owner.get(p_area);
 	ERR_FAIL_COND(!area);
-	area->set_collision_mask(p_mask);
+
+	if (area->get_collision_mask() != p_mask) {
```

Comment:
Can you please perform this operation inside the `set_collision_mask` function?

In this way if any other function change collision mask by calling the `area->set_collision_mask(p_mask);` directly, the same logic is applied.